test/hotspot/jtreg/runtime/cds/appcds/TestCommon.java

```diff
@@ -590,6 +590,11 @@ public static OutputAnalyzer checkExecReturn(OutputAnalyzer output, int ret,
     }
 
 
+    public static String[] concat(ArrayList<String> prefix, String... extra) {
```

Comment:
So you start with an `ArrayList<String>` then convert that to a `String[]` which you pass to a function that then creates a new `ArrayList<String>`, adds all the Strings to it and then converts it to a String[] again! can't help but think there has to be a simpler way to combine an ArrayList with a String[] ??

Reply:
Could use the prefix, add extra to the list then prefix.toArray to avoid such conversion between ArrayList and array.

Reply:
I've removed this code from the current PR. I found other cases where the "runtime prefix" was not passed to child processes. I'll do a more general fix in https://bugs.openjdk.java.net/browse/JDK-8274166